dss/events/handlers/index.py

```diff
@@ -28,6 +28,10 @@ def _index_and_notify(replica: Replica, bundle_fqid: BundleFQID, logger):
         logger.info(f"Received {replica.name} creation event for bundle which will be indexed: {bundle_fqid}")
         document = BundleDocument.from_replica(replica, bundle_fqid, logger)
         index_name = document.prepare_index()
+        versions = document.get_indexed_versions()
+        # No need to remove the document from the index it belongs into
+        versions.pop(index_name, None)
+        document.remove_versions(versions)
```

Comment:
IMO a non-empty set for versions at this point deserves some logging.

Reply:
Good point.